backend/Generator/main.py

```
@@ -360,6 +362,49 @@ def extract_text_from_docx(self, file_path):
+                chunk_files.append(chunk_filename)
+                chunk.export(chunk_filename, format='wav')
+                
+                with sr.AudioFile(chunk_filename) as source:
+                    audio_data = r.record(source)
+                    try:
+                        text = r.recognize_google(audio_data)
+                        full_text.append(text)
+                    except sr.UnknownValueError:
+                        full_text.append("[Unintelligible]")
+                    except sr.RequestError as e:
+                        raise RuntimeError(f"Could not request results; {e}")
```

Ruff (0.14.13)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)